scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,398 @@
+    fluid.log("Processing the snapset Prefs Safes records...");
+    var options = dbLoader.getOptions();
+    var snapSetRecords = JSON.parse(responseString);
+    fluid.each(snapSetRecords.rows, function (aSnapset) {
+        aSnapset.value._deleted = true;
+        options.snapsetPrefsSafes.push(aSnapset.value);
+    });
+    fluid.log("\tSnapset Prefs Safes marked for deletion.");
+    return options.snapsetPrefsSafes;
+};
+
+/**
+ * Find the GPII Key records that are associated with a snapset PrefsSafe, mark
+ * them for deletion, and add them to array of records to delete.
+ * @param {String} responseString - The response from the database query for
+ *                                  retrieving all the GPII Keys.
+ * @return {Array} - The GPII Key records marked for deletion.
+ */
+dbLoader.processGpiiKeys = function (responseString) {
+    fluid.log("Processing the GPII Keys...");
+    var options = dbLoader.getOptions();
+    var gpiiKeyRecords = JSON.parse(responseString);
+    options.gpiiKeys = dbLoader.markPrefsSafesGpiiKeysForDeletion(
+        gpiiKeyRecords, options.snapsetPrefsSafes
+    );
+    fluid.log("\tGPII Keys associated with snapset Prefs Safes marked for deletion.");
+    return options.gpiiKeys;
+};
+
+/**
+ * Given all the GPII Keys records in the database, find the ones that reference
+ * a snapset PrefsSafe.  As each GPII Key is found it is marked for
+ * deletion.
+ * @param {Array} gpiiKeyRecords - Array of GPII Key records from the database.
+ * @param {Array} snapSets - Array of snapset Prefs Safes whose id references
+ *                           its associated GPII Key record.
+ * @return {Array} - the values from the gpiiKeyRecords that are snapset GPII Keys.
+ */
+dbLoader.markPrefsSafesGpiiKeysForDeletion = function (gpiiKeyRecords, snapSets) {
+    var gpiiKeysToDelete = [];
+    fluid.each(gpiiKeyRecords.rows, function (gpiiKeyRecord) {
+        var gpiiKey = fluid.find(snapSets, function (aSnapSet) {
+            if (gpiiKeyRecord.value.prefsSafeId === aSnapSet._id) {
+                return gpiiKeyRecord.value;
+            }
+        }, null);
+        if (gpiiKey !== null) {
+            gpiiKey._deleted = true;
+            gpiiKeysToDelete.push(gpiiKey);
+        }
+    });
+    return gpiiKeysToDelete;
+};
+
+/**
+ * Utility to wrap all the pieces to make a bulk documents deletion request
+ * for the snapset Prefs Safes and their associated GPII keys.  Its intended use
+ * is the parameter of the appropriate promise.then() call.
+ * @param {Object} batchDeleteResponse - the reponse handler configured for the
+ *                                       batch delete request.
+ */
+dbLoader.doBatchDelete = function (batchDeleteResponse) {
+    var options = dbLoader.getOptions();
+    var docsToRemove = options.snapsetPrefsSafes.concat(options.gpiiKeys);
+    var execBatchDelete = dbLoader.createBulkDocsRequest(
+        docsToRemove, batchDeleteResponse, options
+    );
+    execBatchDelete();
```

Comment:
Note that this executes the request blind and provides no way for the caller to wait for conclusion. As a general design note, these should all be functions which return promises.